README.md

```diff
@@ -1,22 +1,381 @@
+
+operators: `==`, `!=`, `>=`, `<=`, `>`, `<`
+
+They evaluate and return logical value of expression they represent.
+> Note that as for now they also support primitive types (booleans, integers, strings) as their operands.
+
+###### Mathematical:
+
+operators: `+`,`-`, `*`, `/`
+
+Those operators return result of mathematical operation evaluated between their operands.
+They only support integers as their operands.
+
+```javascript
+30 + 12;
+84 / 2;
+1 * 42;
+42 - 0;
+```
+
+###### Concatenation
+
+operator: `+`
+
+Adds together two strings and returns the result.
+
+```javascript
+"Hello" + " World!";
+```
+
+###### Number Negation
+
+operator: `-`
+
+Prefixed operator for negating an integer.
+
+```javascript
+52 + -10;
+```
+
+###### Boolean Negation
+
+operator: `!`
+
+Prefixed operator for negating a boolean expression.
+
+```javascript
+const truth = true;
+!truth;
+```
+
+###### Function Call
+
+operators: `()`
+
+To call a function put parenthesis after identifier or any other expression that evaluates to a function literal and pass arguments between them.
+
+```javascript
+const myWeirdFunction = fun(x) {
+    return x + 2;
+};
+
+myWeirdFunction(40);
+
+fun() {
+    print("This is an IIFE!");
+}();
+```
+
+###### Retrieving value with Index
+
+operators: `[]`
+
+The bracket operators are used to retrieve values from arrays and hashes.
+They work just as in any other language.
+
+```javascript
+const myArray = [4, 2, 0];
+
+myArray[0]; // 4
+
+const theUniverse = { 42: "the answer", "isEarthFlat": false };
+
+theUniverse[42];
+theUniverse["isEarthFlat"];
+```
+
+##### Identifiers
+
+Identifiers are also treated as expressions.
+They evaluate to expression they are bind to.
+You can't redeclare a variable inside it's scope but you can overwrite, a identifier that was declared inside scope of one of ancestors of current scope.
+```javascript
+const randomNumber = 40;
+const two = 2;
+
+randomNumber + two;
+```
+
+#### Builtins
+
+Junior have some predefined functions that you can use.
+
+1. `print(values...)` - prints given arguments to the output, returns null.
+2. `len(array|string)` - returns length of argument (array or string).
+3. `first(array)` - returns first element of an array.
+4. `last(array)` - returns last element of given array.
+5. `rest(array)` - returns all the elements of given array but the first one.
+6. `push(array|value)` - returns copy of given array with provided argument as the last element.
+
+
+#### Comments
+
+Junior supports well known single and multi line comments.
+
+```javascript
+// This is a single line comment.
+```
+
+```javascript
+/*
+This is a
+multi line comment.
+ */
+```
+
+> Note: not terminated multi line comment will cause a parsing error.
+
+
+#### Whitespaces
+
+From the interpreter's perspective *whitespaces* are meaningless, but you should always focus on your code's readability.
+
+
+### Error handling
+
+1. Every **Lexical error**, e.g. *invalid token*, stops program parsing.
```

Comment:
from parsing

Reply:
interpreter from parsing or program from being parsed xd

Reply:
interpreter from parsing the program?